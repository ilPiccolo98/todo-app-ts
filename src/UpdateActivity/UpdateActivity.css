.update-form {
  position: relative;
  left: 30%;
}

.float-left-update {
  float: left;
}

.float-right-update {
  float: right;
}

.description-style {
  margin-top: 10px;
}

.checkbox-status-update {
  clear: both;
  padding-top: 15px;
}

.update-button {
  margin-top: 15px;
}

.error-id {
  position: relative;
  left: 35%;
  margin-top: 10px;
}

.error-name {
  position: relative;
  left: 33%;
  margin-top: 10px;
}

.error-description {
  position: relative;
  left: 27%;
  margin-top: 10px;
}
