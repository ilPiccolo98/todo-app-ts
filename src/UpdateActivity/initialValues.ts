export type UpdateActivityValues = {
  id: string;
  name: string;
  description: string;
  status: boolean;
};

const initialValue: UpdateActivityValues = {
  id: "",
  name: "",
  description: "",
  status: false,
};

export default initialValue;
