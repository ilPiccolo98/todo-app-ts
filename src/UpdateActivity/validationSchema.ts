import * as yup from "yup";

const validationSchema = yup.object().shape({
  id: yup.number().required("Id is required"),
  name: yup.string().required("Name is required"),
  description: yup.string().required("Description is required"),
});

export default validationSchema;
