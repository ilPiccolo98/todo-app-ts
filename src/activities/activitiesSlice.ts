import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
import { RootState } from "./activitiesStore";
import Activity from "../types/activity/activity";
import axios, { AxiosError } from "axios";

export enum RetrieveState {
  Error,
  Loading,
  Loaded,
}

export enum AddState {
  Adding,
  Added,
  NotAdded,
  Null,
}

export enum UpdateState {
  Updating,
  Updated,
  NotUpdated,
  Null,
}

export enum DeleteState {
  Deleting,
  Deleted,
  NotDeleted,
  Null,
}

type RetrieveActivities = {
  activities: Activity[];
};

type AddActivity = {
  isItAdded: boolean;
};

type DeleteActivity = {
  isItDeleted: boolean;
};

type UpdateActivity = {
  isItUpdated: boolean;
};

export const fetchAllActivities = createAsyncThunk(
  "activities/fetchAllActivities",
  async (_, thunkAPI) => {
    try {
      const response = await axios.get<RetrieveActivities>(
        "/activities/retrieve"
      );
      return response.data.activities;
    } catch (error) {
      const axiosError = error as AxiosError;
      return thunkAPI.rejectWithValue({ errorMessage: axiosError.message });
    }
  }
);

export const addActivity = createAsyncThunk(
  "activities/addActivity",
  async (
    newActivity: { name: string; description: string; status: boolean },
    thunkAPI
  ) => {
    try {
      const response = await axios.post<AddActivity>("/activities/add", {
        ...newActivity,
      });
      return response.data.isItAdded;
    } catch (error) {
      const axiosError = error as AxiosError;
      return thunkAPI.rejectWithValue({ errorMessage: axiosError.message });
    }
  }
);

export const updateActivity = createAsyncThunk(
  "activities/updateActivity",
  async (activityToUpdate: Activity, thunkAPI) => {
    try {
      const response = await axios.post<UpdateActivity>("/activities/update", {
        ...activityToUpdate,
      });
      return response.data.isItUpdated;
    } catch (error) {
      const axiosError = error as AxiosError;
      return thunkAPI.rejectWithValue({ errorMessage: axiosError.message });
    }
  }
);

export const deleteActivity = createAsyncThunk(
  "activities/deleteActivity",
  async (id: number, thunkAPI) => {
    try {
      const response = await axios.delete<DeleteActivity>(
        `/activities/delete/${id}`
      );
      return response.data.isItDeleted;
    } catch (error) {
      const axiosError = error as AxiosError;
      return thunkAPI.rejectWithValue({ errorMessage: axiosError.message });
    }
  }
);

const activitiesSlice = createSlice({
  name: "activities",
  initialState: {
    activities: new Array<Activity>(),
    addingState: AddState.Null,
    updatingState: UpdateState.Null,
    deletingState: DeleteState.Null,
    retrievingState: RetrieveState.Loading,
  },
  reducers: {
    resetAddState: (state) => {
      state.addingState = AddState.Null;
    },
    resetUpdateState: (state) => {
      state.updatingState = UpdateState.Null;
    },
    resetDeleteState: (state) => {
      state.deletingState = DeleteState.Null;
    },
  },
  extraReducers: (builder) => {
    builder.addCase(addActivity.fulfilled, (state, action) => {
      const isItAdded: boolean = action.payload as boolean;
      if (isItAdded) state.addingState = AddState.Added;
      else {
        state.addingState = AddState.NotAdded;
      }
    });
    builder.addCase(addActivity.pending, (state, action) => {
      state.addingState = AddState.Adding;
    });
    builder.addCase(addActivity.rejected, (state, action) => {
      state.addingState = AddState.NotAdded;
    });
    builder.addCase(updateActivity.fulfilled, (state, action) => {
      const isItUpdated: boolean = action.payload as boolean;
      if (isItUpdated) state.updatingState = UpdateState.Updated;
      else {
        state.updatingState = UpdateState.NotUpdated;
      }
    });
    builder.addCase(updateActivity.pending, (state, action) => {
      state.updatingState = UpdateState.Updating;
    });
    builder.addCase(updateActivity.rejected, (state, action) => {
      state.updatingState = UpdateState.NotUpdated;
    });
    builder.addCase(deleteActivity.fulfilled, (state, action) => {
      const isItDeleted: boolean = action.payload as boolean;
      if (isItDeleted) state.deletingState = DeleteState.Deleted;
      else {
        state.deletingState = DeleteState.NotDeleted;
      }
    });
    builder.addCase(deleteActivity.pending, (state, action) => {
      state.deletingState = DeleteState.Deleting;
    });
    builder.addCase(deleteActivity.rejected, (state, action) => {
      state.deletingState = DeleteState.NotDeleted;
    });
    builder.addCase(fetchAllActivities.fulfilled, (state, action) => {
      state.activities = action.payload as Activity[];
      state.retrievingState = RetrieveState.Loaded;
    });
    builder.addCase(fetchAllActivities.pending, (state, action) => {
      state.retrievingState = RetrieveState.Loading;
    });
    builder.addCase(fetchAllActivities.rejected, (state, action) => {
      state.retrievingState = RetrieveState.Error;
    });
  },
});

export const activitiesSelector = (state: RootState) =>
  state.activities.activities;
export const retrievingStateSelector = (state: RootState) =>
  state.activities.retrievingState;
export const addStateSelector = (state: RootState) =>
  state.activities.addingState;
export const updateStateSelector = (state: RootState) =>
  state.activities.updatingState;
export const deleteStateSelector = (state: RootState) =>
  state.activities.deletingState;

export const {
  resetAddState,
  resetDeleteState,
  resetUpdateState,
} = activitiesSlice.actions;

export default activitiesSlice.reducer;
