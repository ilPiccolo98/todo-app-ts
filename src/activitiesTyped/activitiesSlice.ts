import getInitialState from "./initialActivities";
import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import VectorActivity from "../types/classes/vectorActivity";
import Activity, { ActivityPlain } from "../types/classes/activity";
import { RootState } from "./activitiesStore";

type UpdateActionParameter = {
  id: number;
  name: string;
  description: string;
  status: boolean;
};

type DeleteActionParameter = {
  id: number;
};

type AddActionParameter = {
  name: string;
  description: string;
  status: boolean;
};

const activitiesSlice = createSlice({
  name: "activities",
  initialState: getInitialState(),
  reducers: {
    addActivity: (status, payload: PayloadAction<AddActionParameter>) => {
      const vectorActivity = VectorActivity.fromArrayPlainActivityToVectorActivity(
        status as ActivityPlain[]
      );
      vectorActivity.addActivity(
        new Activity(
          payload.payload.name,
          payload.payload.description,
          payload.payload.status
        )
      );
      status = vectorActivity.toPlainArrayWithPlainActivities();
    },
    deleteActivity: (status, payload: PayloadAction<DeleteActionParameter>) => {
      const vectorActivity = VectorActivity.fromArrayPlainActivityToVectorActivity(
        status as ActivityPlain[]
      );
      vectorActivity.deleteActivity(payload.payload.id);
      status = vectorActivity.toPlainArrayWithPlainActivities();
    },
    updateActivity: (status, payload: PayloadAction<UpdateActionParameter>) => {
      const vectorActivity = VectorActivity.fromArrayPlainActivityToVectorActivity(
        status as ActivityPlain[]
      );
      vectorActivity.updateActivity(
        payload.payload.id,
        payload.payload.name,
        payload.payload.description,
        payload.payload.status
      );
      status = vectorActivity.toPlainArrayWithPlainActivities();
    },
  },
});

export const {
  addActivity,
  deleteActivity,
  updateActivity,
} = activitiesSlice.actions;

export const activitiesSelector = (state: RootState) => state.activities;
export default activitiesSlice.reducer;
