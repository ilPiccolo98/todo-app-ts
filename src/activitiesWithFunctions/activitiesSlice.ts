import { createSlice, PayloadAction } from "@reduxjs/toolkit";
import getInitialActivities from "./initialActivities";
import {
  addActivity as addActivityVector,
  deleteActivity as deleteActivityVector,
  updateActivity as updateActivityVector,
} from "../typesWithoutClasses/vectorActivity/vectorActivity";
import { createActivity } from "../typesWithoutClasses/activity/activity";
import { RootState } from "./activitiesStore";

type AddActivity = {
  name: string;
  description: string;
  status: boolean;
};

type DeleteActivity = {
  id: number;
};

type UpdateActivity = {
  id: number;
  name: string;
  description: string;
  status: boolean;
};

const activitiesSlice = createSlice({
  name: "activities",
  initialState: getInitialActivities(),
  reducers: {
    addActivity: (status, payload: PayloadAction<AddActivity>) => {
      addActivityVector(
        status,
        createActivity(
          payload.payload.name,
          payload.payload.description,
          payload.payload.status
        )
      );
    },
    deleteActivity: (status, payload: PayloadAction<DeleteActivity>) => {
      deleteActivityVector(status, payload.payload.id);
    },
    updateActivity: (status, payload: PayloadAction<UpdateActivity>) => {
      updateActivityVector(
        status,
        payload.payload.id,
        payload.payload.name,
        payload.payload.description,
        payload.payload.status
      );
    },
  },
});

export const {
  addActivity,
  updateActivity,
  deleteActivity,
} = activitiesSlice.actions;

export const activitiesSelector = (state: RootState) => state.activities;

export default activitiesSlice.reducer;
