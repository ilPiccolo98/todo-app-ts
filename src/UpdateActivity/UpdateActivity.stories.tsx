import React from "react";
import { Meta, Story } from "@storybook/react";
import { UpdateActivity, UpdateActivityProps } from "./UpdateActivity";
import store from "../activities/activitiesStore";
import { Provider } from "react-redux";

const meta: Meta = {
  title: "UpdateActivity",
  component: UpdateActivity,
  argTypes: {},
};

export default meta;
const Template: Story<UpdateActivityProps> = (args) => (
  <Provider store={store}>
    <UpdateActivity {...args} />
  </Provider>
);

export const Default = Template.bind({});
Default.args = {};
