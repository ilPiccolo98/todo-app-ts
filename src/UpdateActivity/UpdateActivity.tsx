import React from "react";
import { useSelector, useDispatch } from "react-redux";
import {
  updateActivity,
  activitiesSelector,
  doesActivityExist,
} from "../activities/activitiesSlice";
import { Activity } from "../activities/initialActivities";
import { RootState } from "../activities/activitiesStore";
import initialValues from "./initialValues";
import validationSchema from "./validationSchema";
import Form from "../components/Form/Form";
import Submit from "../components/Submit/Submit";
import TextField from "../components/TextField/TextField";
import CheckBox from "../components/CheckBox/CheckBox";
import "./UpdateActivity.css";
import { useFormik } from "formik";

export interface UpdateActivityProps {}

export const UpdateActivity: React.FC<UpdateActivityProps> = (): JSX.Element => {
  const dispatch = useDispatch();
  const formik = useFormik({
    initialValues,
    validationSchema,
    onSubmit: (values) => {
      if (doesActivityExist(activities, +values.id)) {
        dispatch(
          updateActivity({
            id: +values.id,
            name: values.name,
            description: values.description,
            status: values.status,
          })
        );
      }
    },
  });
  const activities: Array<Activity> = useSelector<RootState, Array<Activity>>(
    activitiesSelector
  );

  return (
    <Form
      variant="primary"
      className="update-form"
      title="Update Activity"
      onSubmit={formik.handleSubmit}
    >
      <div>
        <div className="float-left">
          <TextField
            datatestid="id-updateActivity"
            labelText="Id"
            id="id"
            name="id"
            type="numbers"
            variant="secondary"
            placeholder="Insert Id"
            onChange={formik.handleChange}
            value={formik.values.id}
          />
        </div>
        <div className="float-right">
          <TextField
            datatestid="name-updateActivity"
            labelText="Name"
            id="name"
            name="name"
            type="text"
            variant="secondary"
            placeholder="Insert name"
            onChange={formik.handleChange}
            value={formik.values.name}
          />
        </div>
        <div className="float-left description-style">
          <TextField
            datatestid="description-updateActivity"
            labelText="Description"
            id="description"
            name="description"
            type="text"
            variant="secondary"
            placeholder="Insert description"
            onChange={formik.handleChange}
            value={formik.values.description}
          />
        </div>
        <div className="checkbox-status-update">
          <CheckBox
            datatestid="status-updateActivity"
            className="checkbox-status-update"
            id="status"
            name="status"
            variant="secondary"
            label="Status"
            onChange={formik.handleChange}
            value={formik.values.status}
          />
        </div>
        <Submit
          datatestid="submit-updateActivity"
          className="update-button"
          variant="primary"
        >
          Update
        </Submit>
      </div>
    </Form>
  );
};

export default UpdateActivity;
