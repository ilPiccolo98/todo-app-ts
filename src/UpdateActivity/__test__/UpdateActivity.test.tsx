import React from "react";
import {
  fireEvent,
  render,
  RenderResult,
  waitFor,
  screen,
} from "@testing-library/react";
import UpdateActivity from "../UpdateActivity";
import { Provider } from "react-redux";
import "@testing-library/jest-dom/extend-expect";
import * as actions from "../../activitiesTyped/activitiesSlice";
import store from "../../activities/activitiesStore";

const renderUpdateActivity = () => {
  return render(
    <Provider store={store}>
      <UpdateActivity />
    </Provider>
  );
};

const getUpdateActivityComponents = (
  component: RenderResult<
    typeof import("@testing-library/dom/types/queries"),
    HTMLElement
  >
) => {
  const idTextField = component.getByTestId("id-updateActivity").childNodes[0]
    .childNodes[1] as HTMLElement;
  const nameTextField = component.getByTestId("name-updateActivity")
    .childNodes[0].childNodes[1] as HTMLElement;
  const descriptionTextField = component.getByTestId(
    "description-updateActivity"
  ).childNodes[0].childNodes[1] as HTMLElement;
  const statusCheckbox = component.getByTestId("status-updateActivity")
    .childNodes[0].childNodes[0] as HTMLInputElement;
  const submit = component.getByTestId(
    "submit-updateActivity"
  ) as HTMLInputElement;
  return {
    idTextField,
    nameTextField,
    descriptionTextField,
    statusCheckbox,
    submit,
  };
};

const activitiesSelectorSpy = jest.spyOn(actions, "activitiesSelector");
const updateActivitySpy = jest.spyOn(actions, "updateActivity");

describe("testing UpdateActivity component", () => {
  beforeEach(() => {
    activitiesSelectorSpy.mockReset().mockReturnValue([]);
    updateActivitySpy.mockReset();
  });

  it("should call the updateActivity actions with { id: 1, name: updated, description: updated, status: true }", async () => {
    activitiesSelectorSpy.mockReset().mockReturnValue([
      {
        id: 1,
        name: "activity1",
        description: "activity1",
        status: false,
      },
    ]);
    const component = renderUpdateActivity();
    const {
      idTextField,
      nameTextField,
      descriptionTextField,
      statusCheckbox,
      submit,
    } = getUpdateActivityComponents(component);
    await waitFor(() => {
      fireEvent.change(idTextField, {
        target: {
          value: "1",
        },
      });
    });
    await waitFor(() => {
      fireEvent.change(nameTextField, {
        target: {
          value: "updated",
        },
      });
    });
    await waitFor(() => {
      fireEvent.change(descriptionTextField, {
        target: {
          value: "updated",
        },
      });
    });
    await waitFor(() => {
      fireEvent.click(statusCheckbox);
    });
    await waitFor(() => {
      fireEvent.click(submit);
    });
    expect(updateActivitySpy).toHaveBeenCalledWith({
      id: 1,
      name: "updated",
      description: "updated",
      status: true,
    });
    expect(component).toMatchSnapshot();
  });

  it("shouldn't call the updateActivity actions when the activity doesn't exist", async () => {
    const component = renderUpdateActivity();
    const {
      idTextField,
      nameTextField,
      descriptionTextField,
      statusCheckbox,
      submit,
    } = getUpdateActivityComponents(component);
    await waitFor(() => {
      fireEvent.change(idTextField, {
        target: {
          value: "1",
        },
      });
    });
    await waitFor(() => {
      fireEvent.change(nameTextField, {
        target: {
          value: "updated",
        },
      });
    });
    await waitFor(() => {
      fireEvent.change(descriptionTextField, {
        target: {
          value: "updated",
        },
      });
    });
    await waitFor(() => {
      fireEvent.click(statusCheckbox);
    });
    await waitFor(() => {
      fireEvent.click(submit);
    });
    expect(updateActivitySpy).not.toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });
});
