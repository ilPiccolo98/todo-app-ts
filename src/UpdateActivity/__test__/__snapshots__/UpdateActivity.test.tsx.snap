// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UpdateActivity component should call the updateActivity actions with { id: 1, name: updated, description: updated, status: true } 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form update-form"
      >
        <p>
          <b>
            Update Activity
          </b>
        </p>
        <form
          class=""
        >
          <div>
            <div
              class="float-left"
            >
              <div
                class="secondary-container"
                data-testid="id-updateActivity"
              >
                <label
                  for="id"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Id
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="id-updateActivity"
                    id="id"
                    name="id"
                    placeholder="Insert Id"
                    type="numbers"
                    value="1"
                    variant="secondary"
                  />
                </label>
                <p
                  class="error"
                >
                  <b />
                </p>
              </div>
            </div>
            <div
              class="float-right"
            >
              <div
                class="secondary-container"
                data-testid="name-updateActivity"
              >
                <label
                  for="name"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Name
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="name-updateActivity"
                    id="name"
                    name="name"
                    placeholder="Insert name"
                    type="text"
                    value="updated"
                    variant="secondary"
                  />
                </label>
                <p
                  class="error"
                >
                  <b />
                </p>
              </div>
            </div>
            <div
              class="float-left description-style"
            >
              <div
                class="secondary-container"
                data-testid="description-updateActivity"
              >
                <label
                  for="description"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Description
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="description-updateActivity"
                    id="description"
                    name="description"
                    placeholder="Insert description"
                    type="text"
                    value="updated"
                    variant="secondary"
                  />
                </label>
                <p
                  class="error"
                >
                  <b />
                </p>
              </div>
            </div>
            <div
              class="checkbox-status-update"
            >
              <span
                class="secondary-checkbox"
                data-testid="status-updateActivity"
              >
                <label
                  for="status"
                >
                  <input
                    class="checkbox-status-update"
                    datatestid="status-updateActivity"
                    id="status"
                    label="Status"
                    name="status"
                    type="checkbox"
                    value="true"
                    variant="secondary"
                  />
                  Status
                </label>
              </span>
            </div>
            <input
              class="submit primary-submit update-button"
              data-testid="submit-updateActivity"
              datatestid="submit-updateActivity"
              type="submit"
              value="Update"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form update-form"
    >
      <p>
        <b>
          Update Activity
        </b>
      </p>
      <form
        class=""
      >
        <div>
          <div
            class="float-left"
          >
            <div
              class="secondary-container"
              data-testid="id-updateActivity"
            >
              <label
                for="id"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Id
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="id-updateActivity"
                  id="id"
                  name="id"
                  placeholder="Insert Id"
                  type="numbers"
                  value="1"
                  variant="secondary"
                />
              </label>
              <p
                class="error"
              >
                <b />
              </p>
            </div>
          </div>
          <div
            class="float-right"
          >
            <div
              class="secondary-container"
              data-testid="name-updateActivity"
            >
              <label
                for="name"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Name
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="name-updateActivity"
                  id="name"
                  name="name"
                  placeholder="Insert name"
                  type="text"
                  value="updated"
                  variant="secondary"
                />
              </label>
              <p
                class="error"
              >
                <b />
              </p>
            </div>
          </div>
          <div
            class="float-left description-style"
          >
            <div
              class="secondary-container"
              data-testid="description-updateActivity"
            >
              <label
                for="description"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Description
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="description-updateActivity"
                  id="description"
                  name="description"
                  placeholder="Insert description"
                  type="text"
                  value="updated"
                  variant="secondary"
                />
              </label>
              <p
                class="error"
              >
                <b />
              </p>
            </div>
          </div>
          <div
            class="checkbox-status-update"
          >
            <span
              class="secondary-checkbox"
              data-testid="status-updateActivity"
            >
              <label
                for="status"
              >
                <input
                  class="checkbox-status-update"
                  datatestid="status-updateActivity"
                  id="status"
                  label="Status"
                  name="status"
                  type="checkbox"
                  value="true"
                  variant="secondary"
                />
                Status
              </label>
            </span>
          </div>
          <input
            class="submit primary-submit update-button"
            data-testid="submit-updateActivity"
            datatestid="submit-updateActivity"
            type="submit"
            value="Update"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing UpdateActivity component shouldn't call the updateActivity actions when the activity doesn't exist 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form primary-form update-form"
      >
        <p>
          <b>
            Update Activity
          </b>
        </p>
        <form
          class=""
        >
          <div>
            <div
              class="float-left"
            >
              <div
                class="secondary-container"
                data-testid="id-updateActivity"
              >
                <label
                  for="id"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Id
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="id-updateActivity"
                    id="id"
                    name="id"
                    placeholder="Insert Id"
                    type="numbers"
                    value="1"
                    variant="secondary"
                  />
                </label>
                <p
                  class="error"
                >
                  <b />
                </p>
              </div>
            </div>
            <div
              class="float-right"
            >
              <div
                class="secondary-container"
                data-testid="name-updateActivity"
              >
                <label
                  for="name"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Name
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="name-updateActivity"
                    id="name"
                    name="name"
                    placeholder="Insert name"
                    type="text"
                    value="updated"
                    variant="secondary"
                  />
                </label>
                <p
                  class="error"
                >
                  <b />
                </p>
              </div>
            </div>
            <div
              class="float-left description-style"
            >
              <div
                class="secondary-container"
                data-testid="description-updateActivity"
              >
                <label
                  for="description"
                >
                  <p
                    class="secondary-label"
                  >
                    <b>
                      Description
                    </b>
                  </p>
                  <input
                    class="textField undefined"
                    datatestid="description-updateActivity"
                    id="description"
                    name="description"
                    placeholder="Insert description"
                    type="text"
                    value="updated"
                    variant="secondary"
                  />
                </label>
                <p
                  class="error"
                >
                  <b />
                </p>
              </div>
            </div>
            <div
              class="checkbox-status-update"
            >
              <span
                class="secondary-checkbox"
                data-testid="status-updateActivity"
              >
                <label
                  for="status"
                >
                  <input
                    class="checkbox-status-update"
                    datatestid="status-updateActivity"
                    id="status"
                    label="Status"
                    name="status"
                    type="checkbox"
                    value="true"
                    variant="secondary"
                  />
                  Status
                </label>
              </span>
            </div>
            <input
              class="submit primary-submit update-button"
              data-testid="submit-updateActivity"
              datatestid="submit-updateActivity"
              type="submit"
              value="Update"
            />
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form primary-form update-form"
    >
      <p>
        <b>
          Update Activity
        </b>
      </p>
      <form
        class=""
      >
        <div>
          <div
            class="float-left"
          >
            <div
              class="secondary-container"
              data-testid="id-updateActivity"
            >
              <label
                for="id"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Id
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="id-updateActivity"
                  id="id"
                  name="id"
                  placeholder="Insert Id"
                  type="numbers"
                  value="1"
                  variant="secondary"
                />
              </label>
              <p
                class="error"
              >
                <b />
              </p>
            </div>
          </div>
          <div
            class="float-right"
          >
            <div
              class="secondary-container"
              data-testid="name-updateActivity"
            >
              <label
                for="name"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Name
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="name-updateActivity"
                  id="name"
                  name="name"
                  placeholder="Insert name"
                  type="text"
                  value="updated"
                  variant="secondary"
                />
              </label>
              <p
                class="error"
              >
                <b />
              </p>
            </div>
          </div>
          <div
            class="float-left description-style"
          >
            <div
              class="secondary-container"
              data-testid="description-updateActivity"
            >
              <label
                for="description"
              >
                <p
                  class="secondary-label"
                >
                  <b>
                    Description
                  </b>
                </p>
                <input
                  class="textField undefined"
                  datatestid="description-updateActivity"
                  id="description"
                  name="description"
                  placeholder="Insert description"
                  type="text"
                  value="updated"
                  variant="secondary"
                />
              </label>
              <p
                class="error"
              >
                <b />
              </p>
            </div>
          </div>
          <div
            class="checkbox-status-update"
          >
            <span
              class="secondary-checkbox"
              data-testid="status-updateActivity"
            >
              <label
                for="status"
              >
                <input
                  class="checkbox-status-update"
                  datatestid="status-updateActivity"
                  id="status"
                  label="Status"
                  name="status"
                  type="checkbox"
                  value="true"
                  variant="secondary"
                />
                Status
              </label>
            </span>
          </div>
          <input
            class="submit primary-submit update-button"
            data-testid="submit-updateActivity"
            datatestid="submit-updateActivity"
            type="submit"
            value="Update"
          />
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
